models/classes/test/AssessmentTestXmlBuilder.php

```diff
@@ -0,0 +1,186 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA;
+ *
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoQtiTest\models\test;
+
+use common_exception_Error;
+use common_ext_ExtensionException;
+use DOMDocument;
+use oat\oatbox\service\ConfigurableService;
+use oat\oatbox\service\ServiceNotFoundException;
+use oat\tao\model\service\ApplicationService;
+use qtism\data\AssessmentSection;
+use qtism\data\AssessmentSectionCollection;
+use qtism\data\AssessmentTest;
+use qtism\data\ItemSessionControl;
+use qtism\data\NavigationMode;
+use qtism\data\storage\xml\XmlDocument;
+use qtism\data\storage\xml\XmlStorageException;
+use qtism\data\SubmissionMode;
+use qtism\data\TestPart;
+use qtism\data\TestPartCollection;
+use RuntimeException;
+
+class AssessmentTestXmlBuilder extends ConfigurableService implements AssessmentTestXmlBuilderInterface
```

Comment:
It's a factory, not a builder.

Reply:
IMHO if the intension of the factory is a bit more than creating a single instance based on not complex logic inside it becomes a builder. 
The outcome of this class can be changed/extended with its own extensions, so I'd say it is rather a builder.

Reply:
Builder builds an object step-by-step. Factory creates an object from a single input, encapsulating the logic.

Reply:
ok ok. renamed :)